addons/mail/static/src/models/composer/composer.js

```diff
@@ -682,13 +682,21 @@ function factory(dependencies) {
          * @param {string} mentionKeyword
          */
         _updateSuggestedChannelCommands(mentionKeyword) {
+            let commands = [];
+            if (this.thread) {
+                commands = this.env.messaging.commands.filter(command => {
+                    if (!command.name.includes(mentionKeyword)) {
+                        return false;
+                    }
+                    if (command.channel_types) {
+                        return command.channel_types.includes(this.thread.channel_type);
+                    }
+                    return true;
+                });
+            }
```

Comment:
Do we really need this condition on `this.thread`? Thread should be considered required on composer and not having it is a mistake.

If possible, remove the condition. If there is a crash on a test, ensure the test properly links thread to composer.


-------

If after trying to remove this, there are still issues, I would start with this
```
if (!this.thread) {
    return;
}
```
then use `const` and remove the indent level.

2 reasons for this:
- more readable due to less indent and no `let`
- easier to get rid of the preventive `if` the day we can ensure thread is required on composer (not having a thread on composer should be considered a "mistake")